syntax = "proto3";

package tcatalogue_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message Stop {
	string name = 1;
	Coordinates coords = 2;
}

message Bus {
	string number = 1;
	repeated Stop stops = 2;
	bool is_ring = 3;
}

message Distance {
	string from = 1;
	string to = 2;
	uint32 dist = 3;
}

message TransportCatalogue {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Distance distanses = 3;
	MapSetting map_setting = 4;
	TransportRouter transport_router = 5;
}